webapp/src/modules/store/sagas.ts

```diff
@@ -22,6 +25,18 @@ import {
 export function* storeSaga(client: CatalystClient) {
   yield takeLatest(FETCH_STORE_REQUEST, handleFetchStoreRequest)
   yield takeLatest(UPDATE_STORE_REQUEST, handleUpdateStoreRequest)
+  yield takeEvery(LOCATION_CHANGE, handleLocationChange)
+
+  function* handleLocationChange({
+    payload: { location }
+  }: LocationChangeAction) {
+    const address: string = yield select(getAddress)
+    if (
+      !location.search.includes('section=store_settings&') &&
+      !location.search.includes('viewAsGuest=true')
+    )
+      yield put(revertLocalStore(address))
+  }
```

Comment:
In order to avoid the revert local store action to be logged everytime you could just check if the local store is not null

```suggestion
  function* handleLocationChange({
    payload: { location }
  }: LocationChangeAction) {
    const localStore: Store | null = yield select(getLocalStore)

    if (!localStore) {
      return
    }

    const address: string = yield select(getAddress)
    const allowed = ['section=store_settings&', 'viewAsGuest=true']

    if (!allowed.some(value => location.search.includes(value))) {
      yield put(revertLocalStore(address))
    }
  }
```

Reply:
@fzavalia mm wouldn't it be possible that the localStore is assigned a previous store instead of a null one? according to the reducer:  https://github.com/decentraland/marketplace/blob/master/webapp/src/modules/store/reducer.ts#L103-L111

```
case REVERT_LOCAL_STORE: {
  const { address } = action.payload
  const previous = state.data[address]

  return {
    ...state,
    localStore: previous ? { ...previous } : null
  }
}
```

Tho we could create a selector to know if the local store is different to null and different to all the stores in state:

```ts
export const isLocalStoreDirty = createSelector<
  RootState,
  Record<string, Store>,
  Store | null,
  boolean
>(getData, getLocalStore, (storeData, localStore) => {
  return (
    !!localStore &&
    !Object.values(storeData).some(store => store === localStore)
  )
})
```

And we can use that from within the saga to exit before dispatching the action as you suggest

Reply:
![image](https://user-images.githubusercontent.com/24811313/151008382-949eecab-da4e-470e-8691-bc983a6532cf.png)